tests/unittests/test_firewatch_objects_job.py

```diff
@@ -0,0 +1,220 @@
+                    junit_files.append(file_path)
+
+        assert len(junit_files) > 0
+
+    def test_download_logs(self, tmp_path) -> None:
+        self.logger = logging.getLogger(
+            __name__,
+        )
+
+        files_to_download = ["finished.json", "build-log.txt"]
+
+        job_name = "periodic-ci-windup-windup-ui-tests-v1.1-mtr-ocp4.14-lp-interop-mtr-interop-aws"
+        job_name_safe = "mtr-interop-aws"
+        build_id = "1678283096597729280"
+        storage_client = storage.Client.create_anonymous_client()
+        gcs_bucket = "origin-ci-test"
+        downloads_directory = str(tmp_path)
+
+        logs_path = Job._download_logs(
+            self,
+            downloads_directory=downloads_directory,
+            storage_client=storage_client,
+            gcs_bucket=gcs_bucket,
+            job_name=job_name,
+            build_id=build_id,
+            job_name_safe=job_name_safe,
+        )
+        log_files = []
+
+        for root, dirs, files in os.walk(logs_path):
+            for file_name in files:
+                assert file_name in files_to_download
+                file_path = os.path.join(root, file_name)
+                log_files.append(file_path)
+
+        assert len(log_files) > 0
+
+    def test_find_pod_failures(self, tmp_path) -> None:
+        self.logger = logging.getLogger(
+            __name__,
+        )
+
+        # Set up a logs directory
+        logs_dir = helpers._get_tmp_logs_dir(tmp_path=tmp_path)
+
+        # Test a job with no pod_failures
+        helpers._create_successful_step_pod(logs_dir=logs_dir)
+        pod_failures = Job._find_pod_failures(self, logs_dir=logs_dir)
+        assert len(pod_failures) == 0
+
+        # Test a job with a pod_failure
+        helpers._create_failed_step_pod(logs_dir=logs_dir)
+        pod_failures = Job._find_pod_failures(self, logs_dir=logs_dir)
+        assert len(pod_failures) == 1
+
+    def test_find_test_failures(self, tmp_path) -> None:
+        self.logger = logging.getLogger(
+            __name__,
+        )
+
+        # Set up junit directory
+        junit_dir = helpers._get_tmp_junit_dir(tmp_path=tmp_path)
+
+        # Test job with no test_failures
+        helpers._create_successful_step_junit(junit_dir=junit_dir)
+        failures = Job._find_test_failures(self, junit_dir=junit_dir)
+        assert len(failures) == 0
+
+        # Test job with test_failures
+        helpers._create_failed_step_junit(junit_dir=junit_dir)
+        failures = Job._find_test_failures(self, junit_dir=junit_dir)
+        assert len(failures) == 1
+
+    def test_get_steps(self) -> None:
+        self.logger = logging.getLogger(
+            __name__,
+        )
+        job_name = "periodic-ci-windup-windup-ui-tests-v1.1-mtr-ocp4.14-lp-interop-mtr-interop-aws"
+        job_name_safe = "mtr-interop-aws"
+        build_id = "1678283096597729280"
+        storage_client = storage.Client.create_anonymous_client()
+        gcs_bucket = "origin-ci-test"
```

Comment:
It is, it's the only way we can test this function. I am fine removing the test as I am still a bit scared of these logs being removed eventually. That being said, the test will not modify anything in the job itself and if the logs do get removed, we just need to update the variables to a more recent job.

Reply:
I think I will just remove this test for now as it could cause some issues. I like the idea of finding the latest run from one of our jobs, but if the list of steps change, this test will start to fail. I'm not super worried about this function anyway, it has never given me any issues.